src/compiler/emitter.ts

```diff
@@ -3958,16 +3955,14 @@ namespace ts {
         function emitSourceFile(node: SourceFile) {
             writeLine();
             const statements = node.statements;
-            if (emitBodyWithDetachedComments) {

```

Comment:
I'm not really sure what this check was suppose to be. It (and another one like it) are always true, because the function always exists. Isn't this something the checker can figure out?

Reply:
Huh, interesting. I guess I've definitely written code like `globalThis?.navigator?.something?.somethingElse` to be defensive because the same code is shared in multiple contexts, and I didn't want it to complain.

Reply:
Several years ago comment emit was in a separate file and this function was only conditionally available. We merged comment emit back in for performance reasons, but these conditions were unintentionally left in.